packages/wallet-sdk/src/core/communicator/util.test.ts

```diff
@@ -0,0 +1,91 @@
+import { closePopup, openPopup } from './util';
+
+describe('Communicator util', () => {
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe('openPopup', () => {
+    it('opens a popup window', () => {
+      const url = new URL('https://keys.coinbase.com');
+      const mockFocus = jest.fn();
+
+      const windowSpy = jest.spyOn(window, 'open').mockImplementation(() => {
+        return {
+          focus: mockFocus,
+        } as any;
+      });
+
+      openPopup(url);
+
+      expect(windowSpy).toHaveBeenCalledTimes(1);
+      expect(windowSpy).toHaveBeenCalledWith(url, 'Smart Wallet', expect.any(String));
+      expect(mockFocus).toHaveBeenCalled();
+    });
+
+    it('uses the correct dimensions and location for the popup window', () => {
+      const url = new URL('https://keys.coinbase.com');
+
+      const mockScreenXY = 100;
+      const mockScreenInnerWidthHeight = 50;
+      Object.defineProperty(window, 'screenX', { value: mockScreenXY });
+      Object.defineProperty(window, 'screenY', { value: mockScreenXY });
+      Object.defineProperty(window, 'innerWidth', { value: mockScreenInnerWidthHeight });
+      Object.defineProperty(window, 'innerHeight', { value: mockScreenInnerWidthHeight });
+
+      const windowSpy = jest.spyOn(window, 'open').mockImplementation(() => {
+        return {
+          focus: jest.fn(),
+        } as any;
+      });
+
+      openPopup(url);
+
+      const expectedWidth = 420;
+      const expectedHeight = 540;
+      const expectedLeft = (mockScreenInnerWidthHeight - expectedWidth) / 2 + mockScreenXY;
+      const expectedTop = (mockScreenInnerWidthHeight - expectedHeight) / 2 + mockScreenXY;
```

Comment:
![image](https://github.com/coinbase/coinbase-wallet-sdk/assets/966662/1deab3f1-6a34-4b0d-9618-ba4e3d136616)
this is just repetition of the same code